sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuJsonReadCommon.scala

```diff
@@ -336,6 +336,8 @@ object GpuJsonReadCommon {
     .withLeadingZeros(options.allowNumericLeadingZeros)
     .withNonNumericNumbers(options.allowNonNumericNumbers)
     .withUnquotedControlChars(allowUnquotedControlChars)
+    .withCudfPruneSchema(true)
+    .withExperimental(true)
```

Comment:
curious on what the plan is for the `withExperimental` flag to be deprecated. 

As it is here, I don't know what it does, but the cuDF code does say that it enables features such as utf-8 and new column tree construction. Is that the extent? Should we add a comment here that says why we set it?

Reply:
The experimental flag enables most of the fixes. They are things that should be okay to work with the python side of things, but to reduce the risk for the 24.10 release they were put under this flag. It is likely that it will be removed at some point in 24.12, but we can wait and see